format/nexus.py

```diff
@@ -1395,6 +1395,9 @@ def detectorgroupdatafactory(obj, instrument):
         if key.startswith("_filename_"):
             continue
 
+        if "signal" in obj.handle.attrs and key != obj.handle.attrs["signal"]:
```

Comment:
Since `nexus.py` is a quagmire of mysteries at the best of time, would welcome a comment to effect as to what this is doing, for future reference.

Reply:
Worth adding an assert that the `signal` data set _is_ present?

Reply:
If it's not present, won't that throw an error later on?

FWIW signal is part of the NeXus spec so we _think_ we aren't doing anything wrong by having multiple NXdata blocks where the default is identified by `@signal`.  But a comment here wouldn't be amiss that `@signal` accounts for this use cases like this.

Reply:
In this case, obj is the dataset, and if signal is not an attribute of that dataset, then it shouldn't actually execute the ```key != obj.handle.attrs["signal"]``` bit . So I think we are guarded by the case where signal is not an attribute of data  ?

Reply:
I think that's right, @dermen